BASE/src/ServerTelemetryChannel/Implementation/TransmissionSender.cs

```diff
@@ -155,7 +197,9 @@ private async Task StartSending(Transmission transmission)
                 try
                 {
                     TelemetryChannelEventSource.Log.TransmissionSendStarted(acceptedTransmission.Id);
-                    responseContent = await acceptedTransmission.SendAsync().ConfigureAwait(false);
+                    transmissionTask = acceptedTransmission.SendAsync();
+                    this.inFlightTransmissions.TryAdd(transmission.FlushAsyncId, transmissionTask);
```

Comment:
adding to the dictionary can be done *before* called SendAsync() to ensure its always added to dictionary, irrespective of any exceptions.

Reply:
`transmissionTask` is generated by `SendAsync()`. Only awaiting on a task will throw an exception. Before awaiting a task, an entry is added to dictionary. So irrespective of exception it gets added to dictionary.